Add support for bandwidth sensors in Glances

yuvalabou opened the pull request:
<!--
  You are amazing! Thanks for contributing to our project!
  Please, DO NOT DELETE ANY TEXT from this template! (unless instructed).
-->
## Breaking change
<!--
  If your PR contains a breaking change for existing users, it is important
  to tell them what breaks, how to make it work again and why we did this.
  This piece of text is published with the release notes, so it helps if you
  write it towards our users, not us.
  Note: Remove this section if this PR is NOT a breaking change.
-->
This may drop API version 2?

## Proposed change
<!--
  Describe the big picture of your changes here to communicate to the
  maintainers why we should accept this pull request. If it fixes a bug
  or resolves a feature request, be sure to link to that issue in the
  additional information section.
-->
Add Tx and Rx sensors for present network interfaces.

Still work in progress and I might need some help -  for example:
    Should we round the value and represent them as MB?
    Add more none realtime traffic (e.g commulative)?

Also - I have zero clue on how to create test for it so..

## Type of change
<!--
  What type of change does your PR introduce to Home Assistant?
  NOTE: Please, check only 1! box!
  If your PR requires multiple boxes to be checked, you'll most likely need to
  split it into multiple PRs. This makes things easier and faster to code review.
-->

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New integration (thank you!)
- [x] New feature (which adds functionality to an existing integration)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information
<!--
  Details are important, and help maintainers processing your PR.
  Please be sure to fill out additional details, if applicable.
-->

- This PR fixes or closes issue: fixes #
- This PR is related to issue: 
- Link to documentation pull request:  https://github.com/home-assistant/home-assistant.io/pull/18419

## Checklist
<!--
  Put an `x` in the boxes that apply. You can also fill these out after
  creating the PR. If you're unsure about any of them, don't hesitate to ask.
  We're here to help! This is simply a reminder of what we are going to look
  for before merging your code.
-->

- [x] The code change is tested and works locally.
- [ ] Local tests pass. **Your PR cannot be merged unless tests pass**
- [x] There is no commented out code in this PR.
- [ ] I have followed the [development checklist][dev-checklist]
- [x] The code has been formatted using Black (`black --fast homeassistant tests`)
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

If the code communicates with devices, web services, or third-party tools:

- [ ] The [manifest file][manifest-docs] has all fields filled out correctly.  
      Updated and included derived files by running: `python3 -m script.hassfest`.
- [ ] New or updated dependencies have been added to `requirements_all.txt`.  
      Updated by running `python3 -m script.gen_requirements_all`.
- [ ] Untested files have been added to `.coveragerc`.

The integration reached or maintains the following [Integration Quality Scale][quality-scale]:
<!--
  The Integration Quality Scale scores an integration on the code quality
  and user experience. Each level of the quality scale consists of a list
  of requirements. We highly recommend getting your integration scored!
-->

- [ ] No score or internal
- [ ] 🥈 Silver
- [ ] 🥇 Gold
- [ ] 🏆 Platinum

<!--
  This project is very active and we have a high turnover of pull requests.

  Unfortunately, the number of incoming pull requests is higher than what our
  reviewers can review and merge so there is a long backlog of pull requests
  waiting for review. You can help here!
  
  By reviewing another pull request, you will help raise the code quality of
  that pull request and the final review will be faster. This way the general
  pace of pull request reviews will go up and your wait time will go down.
  
  When picking a pull request to review, try to choose one that hasn't yet
  been reviewed.

  Thanks for helping out!
-->

To help with the load of incoming pull requests:

- [ ] I have reviewed two other [open pull requests][prs] in this repository.

[prs]: https://github.com/home-assistant/core/pulls?q=is%3Aopen+is%3Apr+-author%3A%40me+-draft%3Atrue+-label%3Awaiting-for-upstream+sort%3Acreated-desc+review%3Anone

<!--
  Thank you for contributing <3

  Below, some useful links you could explore:
-->
[dev-checklist]: https://developers.home-assistant.io/docs/en/development_checklist.html
[manifest-docs]: https://developers.home-assistant.io/docs/en/creating_integration_manifest.html
[quality-scale]: https://developers.home-assistant.io/docs/en/next/integration_quality_scale_index.html
[docs-repository]: https://github.com/home-assistant/home-assistant.io

yuvalabou committed:
Add support for bandwidth sensors in Glances

yuvalabou committed:
Add support for bandwidth sensors in Glances

pvizeli committed:
Python 3.9.6 / Base image 2021.07.0 (#52605)

Commit:
[ci skip] Translation update

engrbm87 commented:
Hello @yuvalabou,
Are these network interfaces available in glances_api v2? You mentioned a breaking change but I didn't understand it.
As for the value representation I think it should be minimum in KB and you can add an option to specify the unit of measurement for for these entities.

yuvalabou commented:
> Are these network interfaces available in glances_api v2? You mentioned a breaking change but I didn't understand it.

I actually don't know, couldn't find any docs for the v2 API and I can't test it locally..

> As for the value representation I think it should be minimum in KB and you can add an option to specify the unit of measurement for these entities.

Sure.

Adminiuga committed:
Bump up ZHA dependencies (#52611)

bdraco committed:
Fix deadlock at shutdown with python 3.9 (#52613)

frenck committed:
Fix broadlink creating duplicate unique IDs (#52621)

firstof9 committed:
Fix mysensors rgb light (#52604)

* remove assert self._white as not all RGB will have a white channel

* suggested change

* Update homeassistant/components/mysensors/light.py

Co-authored-by: Erik Montnemery <erik@montnemery.com>

Co-authored-by: Franck Nijhof <frenck@frenck.nl>
Co-authored-by: Erik Montnemery <erik@montnemery.com>

bramkragten committed:
Update frontend to 20210707.0 (#52624)

yuvalabou committed:
change network uom

yuvalabou committed:
change network uom

yuvalabou commented:
Made it MB as kb usually not that useful (my humble opinion) and we generally do not need that resolution..
Don't know yet how to make it configurable, as we need another step in the config flow

pvizeli committed:
Bump opencv to 4.5.2.54 (#52630)

anaisbetts committed:
Fix service registration typo in Nuki integration (#52631)

ludeeus committed:
Handle KeyError when accessing device information (#52650)

chemelli74 committed:
Fix Fritz default consider home value (#52648)

thecode committed:
Fix Waze Travel Time tests (#52663)

allenporter committed:
Refactor decompression timestamp validation logic in stream component (#52462)

* Refactor dts validation logic into a separate function

Create a decompression timestamp validation function to move the logic out of
the worker into a separate class. This also uses the python itertools.chain
to chain together the initial packets with the remaining packets in the
container iterator, removing additional inline if statements.

* Reset dts validator when container is reset

* Fix typo in a comment

* Reuse existing dts_validator when disabling audio stream

bachya committed:
Add missing type annotations to Guardian (#52598)

bachya committed:
Add missing type annotations to Notion (#52599)

bachya committed:
Add missing type annotations to ReCollect Waste (#52600)

bachya committed:
Bump simplisafe-python to 11.0.1 (#52684)

Commit:
[ci skip] Translation update

tkdrob committed:
Use entity class attributes for anel_pwrctrl (#52594)

* Use entity class attributes for anel_pwrctrl

* Tweak

esev committed:
pyWeMo version bump (0.6.5) (#52701)

jjlawren committed:
Warn if `interface_addr` remains in Sonos configuration (#52652)

tkdrob committed:
Use entity class attributes for avion (#52696)

uvjustin committed:
Move recorder.py import to runtime (#52682)

tkdrob committed:
Use entity class attributes for aurora_abb_power (#52692)

cgtobi committed:
Use class properties in netatmo (#52705)

tkdrob committed:
Use entity class attributes for Aten pe (#52687)

* Use entity class attributes for aten_pe

* Use entity class attributes for atag

avee87 committed:
Use iso-formatted times in MetOffice weather forecast (#52672)

* Fixed raw datetime in MetOffice weather forecast

* Use datetime in sensor attribute

frenck committed:
Ensure Forecast.Solar returns an iso formatted timestamp (#52669)

alengwenus committed:
Hint for str type instead of explicitly casting to str (#52712)

agners committed:
Fix precipitation calculation for hourly forecast (#52676)

It seems that hourly forecast have precipitation in 3h blocks.

tkdrob committed:
Use entity class attributes for anthemav (#52602)

tkdrob committed:
Use entity class attributes for apcupsd (#52662)

jjlawren committed:
Ignore unused keys from Sonos device properties callback (#52660)

* Ignore known but unused keys from device callback

* Fix bug, add test

janiversen committed:
Add check for _client existence in modbus (#52719)

jesserockz committed:
Esphome fix camera image (#52738)

nielstron committed:
Upgrade Fronius dependency to 0.5.3 (#52737)

supports more values of new Gen24 type of fronius device

Tommatheussen committed:
Fix KNX Fan features (#52732)

* Fan entity should return support features

* Revert "Fan entity should return support features"

This reverts commit 3ad0e87708fbf1847aaa26e3bc76fcac365a1640.

* Restore supported_features for KNX fan

tkdrob committed:
Use entity class attributes for aurora (#52690)

* Use entity class attributes for aurora

* fix

JonGilmore committed:
Bump pylutron to 0.2.8 fixing python 3.9 incompatibility (#52702)

apaperclip committed:
Remove scale calculation for climacell cloud cover (#52752)

Jc2k committed:
Fix homebridge devices becoming unavailable frequently (#52753)

Update to aiohomekit 0.4.3 and make sure service type UUID is normalised before comparison

Co-authored-by: J. Nick Koston <nick@koston.org>

Commit:
[ci skip] Translation update

bdr99 committed:
Upgrade pymazda to 0.2.0 (#52775)

OttoWinter committed:
Fix ESPHome Camera not merging image packets (#52783)

Jc2k committed:
Support certain homekit devices that emit invalid JSON (#52759)

Kane610 committed:
Bump dependency to properly handle current and voltage not being reported on some zhapower endpoints (#52764)

stefano055415 committed:
Add device info to Freedompro (#52715)

* Update Freedompro

* Update homeassistant/components/freedompro/light.py

Co-authored-by: Franck Nijhof <frenck@frenck.nl>

Co-authored-by: Franck Nijhof <frenck@frenck.nl>

firstof9 committed:
Add transition support to zwave_js lights (#52160)

* Add transition support to zwave_js lights

* Add transition support to color_switch lights

* simplify and add tests

* fix logic

* add check for color transition to add SUPPORT_TRANSITON supported features

* Use new metadata property

* Use new metadata property

* update tests and device state dump json files

* fix file perms

* update tests and fixtures with new metadata

* update test

* update test

* update tests for color transitions

* check for color tansitions as well

* more tests

* fix color transtions

* remove unneed default

* set add_to_watched_value_ids to false

* set transition default

* properly set default

* update tests

* make sure transition is an int

* suggested changes

* Update homeassistant/components/zwave_js/light.py

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

* Update homeassistant/components/zwave_js/light.py

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

* Update homeassistant/components/zwave_js/light.py

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

* Update homeassistant/components/zwave_js/light.py

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

* Update homeassistant/components/zwave_js/light.py

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

* formatting

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

chemelli74 committed:
Fix Neato parameter for token refresh (#52785)

* Fix param

* cleanup

Commit:
[ci skip] Translation update

Adminiuga committed:
Bump up ZHA depdencies (#52818)

elupus committed:
Update arcam lib to 0.7.0 (#52829)

cdce8p committed:
Fix pylint issue with stream component c-extension (#52847)

* Rename 'extension-pkg-whitelist' setting to 'extension-pkg-allow-list'
* Add 'av.stream' and 'av.audio.stream'
* Replace 'Any' type hint

Commit:
[ci skip] Translation update

balloob committed:
Rename preview task to run (#52857)

raman325 committed:
Bump zwave-js-server-python to 0.27.1 (#52885)

ludeeus committed:
Bump pyhaversion to 21.7.0 (#52880)

tkdrob committed:
Use entity class attributes for bbb_gpio (#52837)

tkdrob committed:
Use entity class attributes for bayesian (#52831)

tkdrob committed:
Use entity class attributes for Beewi smartclim (#52839)

* Use entity class attributes for beewi_smartclim

* rework

tkdrob committed:
Use entity class attributes for azure_devops (#52698)

tkdrob committed:
Use entity class attributes for arlo (#52681)

* Use entity class attributes for arlo

* revert sensor

EddyK69 committed:
Add AllTrips sensors for BMW Connected Drive (#50420)

* Add AllTrips sensors for BMW Connected Drive

Added several new AllTrips sensors and some optional extra AllTrips sensors (disabled by default)

* Fix for failed checks

* Fix for failed check (black)

* Code tidying

Changed code after useful comments ;)

raman325 committed:
Add siren platform (#48309)

* Add siren platform

* add more supported flags and an ability to set siren duration

* tone can be int or string

* fix typing

* fix typehinting

* fix typehints

* implement a proposed approach based on discussion

* Address comments

* fix tests

* Small fix

* Update homeassistant/components/demo/siren.py

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

* Update homeassistant/components/demo/siren.py

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

* Update homeassistant/components/demo/siren.py

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

* Update homeassistant/components/demo/siren.py

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

* Update homeassistant/components/demo/siren.py

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

* typing

* use class attributes

* fix naming

* remove device from service description

* Filter out params from turn on service

* fix tests

* fix bugs and tests

* add test

* Combine is_on test with turn on/off/toggle service tests

* Update homeassistant/components/siren/__init__.py

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

* fix filtering of turn_on attributes

* none check

* remove services and attributes for volume level, default duration, and default tone

* Update homeassistant/components/siren/__init__.py

Co-authored-by: Franck Nijhof <frenck@frenck.nl>

* Update homeassistant/components/siren/__init__.py

Co-authored-by: Franck Nijhof <frenck@frenck.nl>

* Update homeassistant/components/siren/__init__.py

Co-authored-by: Franck Nijhof <frenck@frenck.nl>

* import final

* Update homeassistant/components/siren/__init__.py

Co-authored-by: Franck Nijhof <frenck@frenck.nl>

* Fix typing and used TypedDict for service parameters

* remove is_on function

* remove class name redundancy

* remove extra service descriptions

* switch to positive_int

* fix schema for tone

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>
Co-authored-by: Franck Nijhof <frenck@frenck.nl>

bdraco committed:
Send ssdp requests to ipv4 broadcast as well (#52760)

* Send ssdp requests to 255.255.255.255 as well

- This matches pysonos behavior and may fix reports of
  inability to discover some sonos devices
  https://github.com/amelchio/pysonos/blob/master/pysonos/discovery.py#L120

* Update homeassistant/components/ssdp/__init__.py

LeszekSwirski committed:
Add device classes to homematicip_cloud cover (#52793)

Make HMIP covers report a SHUTTER/BLIND/GARAGE device_class (as
appropriate).

Commit:
[ci skip] Translation update

raman325 committed:
Initial support for zwave_js device conditions (#52003)

scop committed:
Implement stable unique id for Huawei LTE, requires credentials on setup (#49878)

janiversen committed:
Add more data_types to modbus (#52423)

* Add more data_types.

* Use new struct when writing temperature.

Kane610 committed:
Prefer using xy over hs when supported by light (#52883)

chemelli74 committed:
Fix Soundbar exclusion from SamsungTV (#51023)

* Improved check

* Fix tests

* Fix logic and tests

* Update tests

tkdrob committed:
Use entity class attributes for Blinksticklight (#52892)

* Use entity class attributes for blinksticklight

* rework

* remove self._serial

tkdrob committed:
Use entity class attributes for Bh1750 (#52886)

* Use entity class attributes for bh1750

* rework

cdce8p committed:
Import Protocol from typing (#52848)

tkdrob committed:
Use entity class attributes for Bizkaibus (#52888)

dependabot[bot] committed:
Bump dessant/lock-threads from 2.0.3 to 2.1.1 (#52899)

Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

Danielhiversen committed:
Surepetcare, fix set_lock_state (#52912)

Signed-off-by: Daniel Hjelseth Høyer <github@dahoiv.net>

tkdrob committed:
Use entity class attributes for Bitcoin (#52887)

* Use entity class attributes for bitcoin

* rework

* fix

* tweak

bdraco committed:
Bump aiohomekit to 0.5.1 to solve performance regression (#52878)

- Changelog: https://github.com/Jc2k/aiohomekit/compare/0.5.0...0.5.1

- Note that #52759 will need to be cherry-picked under this commit

bdraco committed:
Add the Trane brand to nexia (#52805)

TomBrien committed:
Improve testing of option flow in Coinbase (#52870)

jjlawren committed:
Ignore Sonos Boost devices during discovery (#52845)

cgtobi committed:
Add some type hints for webhook component (#52895)

* Add some type hints

* Fix type hint

* Address comment

* Make pylint happy

bdraco committed:
Handle dhcp packets without a hostname (#52882)

* Handle dhcp packets without a hostname

- Since some integrations only match on OUI we want to
  make sure they still see devices that do not request
  a specific hostname

* Update tests/components/dhcp/test_init.py

* Update homeassistant/components/dhcp/__init__.py

Co-authored-by: Franck Nijhof <git@frenck.dev>

Co-authored-by: Franck Nijhof <git@frenck.dev>

bdraco committed:
Add zeroconf discovery to Sonos (#52655)

emontnemery committed:
Improve typing of Tasmota (2/3) (#52747)

* Improve typing of Tasmota (2/3)

* Add more typing, add TasmotaOnOffEntity

* Address review comments

tkdrob committed:
Use entity class attributes for apple_tv (#52664)

* Use entity class attributes for apple_tv

* fix pylint

* tweak

emontnemery committed:
Set device_class on temperature sensors R-Z (#52920)

emontnemery committed:
Improve typing of Tasmota (3/3) (#52748)

teharris1 committed:
Bump pyinsteon to 1.0.11 (#52927)

frenck committed:
Upgrade sentry-sdk to 1.3.0 (#52926)

janiversen committed:
Correct Wrong "raise" in modbus validators. (#52924)

emontnemery committed:
Improve typing of Tasmota (1/3) (#52746)

cyberjunky committed:
Bump python-fireservicerota to 0.0.42 (#52807)

janiversen committed:
Add array write to turn_on/off in modbus switch/fan/light (#52582)

emontnemery committed:
Set device_class on temperature sensors F-K (#52918)

* Set device_class on temperature sensors F-K

* Fix juicenet sensor

milanmeu committed:
Use properties instead of raw data in the rituals integration (#52587)

emontnemery committed:
Set device_class on temperature sensors L-Q (#52919)

bwduncan committed:
Poll Nissan servers for battery updates (#44826)

bdraco committed:
Fix nexia thermostats humidify without dehumidify support (#52758)

bdraco committed:
Fix recorder purge with sqlite3 < 3.32.0 (#52929)

emontnemery committed:
Set device_class on temperature sensors A-E (#49524)

Co-authored-by: Franck Nijhof <git@frenck.dev>

tkdrob committed:
Use entity class attributes for bbox (#52838)

* Use entity class attributes for bbox

* tweak

tkdrob committed:
Use entity class attributes for Blockchain (#52894)

* Use entity class attributes for blockchain

* rework

* tweak

jjlawren committed:
Bump pysonos to 0.0.52 (#52934)

tkdrob committed:
Use entity class attributes for Blebox (#52890)

* Use entity class attributes for blebox

* rework

* Apply suggestions from code review

Co-authored-by: Franck Nijhof <git@frenck.dev>

Co-authored-by: Franck Nijhof <git@frenck.dev>

OttoWinter committed:
ESPHome enable static type checking (#52348)

jjlawren committed:
Use entity class attributes for Plex (#52617)

scop committed:
String formatting cleanups (#52937)

tkdrob committed:
Use entity class attributes for aqualogic (#52668)

cgtobi committed:
Add schedule selector for Netatmo (#52909)

Co-authored-by: Joakim Sørensen <hi@ludeeus.dev>

GitHub-Action committed:
[ci skip] Translation update

raman325 committed:
Fix siren turn on parameter filtering (#52947)

* Fix siren turn on parameter filtering

* fix test

raman325 committed:
Handle FailedCommand exceptions in zwave_js WS API (#52461)

* Handle zwave-js errors in WS API

* Unsubscribe callbacks when zwave-js error is caught

* fix tests

* simplify unsub logic

* add tests

* add kwargs to be safe

* use existing msg format

* switch to generic failed command handling

* remove unneeded constant

* Update homeassistant/components/zwave_js/api.py

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

* Update homeassistant/components/zwave_js/api.py

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

* fix

Co-authored-by: Martin Hjelmare <marhje52@gmail.com>

raman325 committed:
Change behavior of Z-Wave JS services (#52941)

* Change behavior of Z-Wave JS services

* pop parameters in for loop

* Update logger message

emontnemery committed:
Minor test coverage improvement of mfi and zwave sensors (#52935)

stefano055415 committed:
Add switch support to Freedompro (#52727)

* Update Freedompro

* add test state updates from the API

* fix test switch

* fix test

bdraco committed:
Expose async_get_source_ip in the network integration (#52901)

* Expose async_get_source_ip in the network integration

* Handle source ip on disabled interface

* add coverage

raman325 committed:
Switch to using entry.async_on_remove (#52952)

emontnemery committed:
Return empty when listing statistic_ids for unsupported statistic (#52954)

Kirchoff committed:
Fix modbus climate precision configuration variable (#52651)

* Updated precision to follow the tenths, halves, whole notation used by
other home assistant climate modules. Added the precision @property so
that home assistant can handle this rounding in the frontend, rather
than in the _async_read_register() method.

* Fixed a pylinter error for periods in user-facing log messages, and
updated `precision` defaults in components/modbus/__init__.py to be
consistent with an error case, using `PRECISION_WHOLE`.

* revert changes to `precision:` configuration variable
instead, the climate `precision()` function will infer whether or not
to display in whole or tenths. halves will be unsupported, which should
be fine.

* re-added missing line that was removed

* revert change to use self._input_type instead of CALL_TYPE_REGISTER_HOLDING

raman325 committed:
Validate tone is valid when processing siren.turn_on service call (#52953)

* Validate tone is valid when processing siren.turn_on service call

* Better message

emontnemery committed:
Improve docstring for async_get_device_class_lookup (#52921)

yuvalabou committed:
Merge branch 'glances-network' of https://github.com/yuvalabou/core into glances-network

homeassistant commented:
Hi @GitHub-Action, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;core#52603).

Once you do that we will be able to review and accept this pull request.

Thanks!

yuvalabou commented:
Rebasing failed :(
